### ISSUE TITLE
salt-minion and salt-master displaying traceback when started

### ISSUE THREAD

**Issue by nmadhok** [saltstack/salt#26672]
When starting/restarting salt-master/salt-minion on latest develop branch, it displays a traceback and gets hung. After a KeyBoard Interrupt, it displays ``OK``. 

```
root@nitin-salt-develop:~/salt$ service salt-master stop
Stopping salt-master daemon:                               [  OK  ]
root@nitin-salt-develop:~/salt$ service salt-master start
Starting salt-master daemon: Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
Error in sys.exitfunc:
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/log/setup.py", line 906, in __global_logging_exception_handler
    shutdown_multiprocessing_logging_listener()
  File "/usr/lib/python2.6/site-packages/salt/log/setup.py", line 800, in shutdown_multiprocessing_logging_listener
    __MP_LOGGING_QUEUE_PROCESS.terminate()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 111, in terminate
    self._popen.terminate()
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 139, in terminate
    os.kill(self.pid, signal.SIGTERM)
OSError: [Errno 3] No such process

Original exception was:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
                                                           [  OK  ]
```

**Comment by jfindlay** [comment]
@nmadhok, does this happen if you create a new VM and install develop?

**Comment by nmadhok** [comment]
@jfindlay Haven't tried that yet. But i don't see this problem on 2015.8 or 2015.5 or 2014.7 branch on the same server.

**Comment by cachedout** [comment]
I cannot replicate this at all and the test suite sees no indication of this happening on any platform currently. If we can get clear steps to reproduce this, we can certainly re-open it.

**Comment by nmadhok** [comment]
@cachedout Deployed a fresh new server and was able to replicate this:

```
root@cu-inf-d-slt01:~/salt$ service salt-master start
Starting salt-master daemon: /usr/lib64/python2.6/site-packages/Crypto/Util/number.py:57: PowmInsecureWarning: Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.
  _warn("Not using mpz_powm_sec.  You should rebuild using libgmp >= 5 to avoid timing attack vulnerability.", PowmInsecureWarning)
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 117, in join
    assert self._parent_pid == os.getpid(), 'can only join a child process'
AssertionError: can only join a child process


^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib64/python2.6/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/lib64/python2.6/multiprocessing/util.py", line 269, in _exit_function
    p.join()
  File "/usr/lib64/python2.6/multiprocessing/process.py", line 119, in join
    res = self._popen.wait(timeout)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 122, in wait
    return self.poll(0)
  File "/usr/lib64/python2.6/multiprocessing/forking.py", line 107, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
Error in sys.exitfunc:
                                                           [  OK  ]
```

**Comment by richardhansen** [comment]
I believe this is the same bug as #25985.

**Comment by terminalmage** [comment]
@nmadhok Can you be more specific about the changeset that you found which seems to cause this?

**Comment by nmadhok** [comment]
This issue doesn't exist on 0a499bd0fa4de916c181d5c894eefcc515dea75e but started appearing after 0fa0cc003dd521e7cad2ff1b33af257a6817056f

<img width="923" alt="screen shot 2015-09-08 at 1 05 15 pm" src="https://cloud.githubusercontent.com/assets/3374962/9741802/6e49462c-562a-11e5-849c-c3ef44c41e52.png">

Do a git checkout of the two and try to restart the services.

**Comment by terminalmage** [comment]
@nmadhok I did my own git bisect and found a very different bad commit, one that looks more like it has something to do with this issue: https://github.com/saltstack/salt/commit/25ef141536c4ce0c3681b03e389719d5a9f7d495

@s0undt3ch can you have a look?

**Comment by nmadhok** [comment]
@terminalmage I did not do a git bisect. Just checkout out the respective commits and tried to restart services. Noticed the issue arising after the commit I mentioned above.

**Comment by terminalmage** [comment]
OK, well my commit had nothing to do with multiprocessing, so it looks like you may have made a mistake. I've notified @s0undt3ch about the offending commit.

**Comment by nmadhok** [comment]
@terminalmage I agree that your commits had nothing to do with multiprocessing. I am able to still reproduce it though after checking out the above mentioned commit and it doesn't exist on the commit before that. It looks like it may be related to the change made by @s0undt3ch

**Comment by cachedout** [comment]
@nmadhok I'm still at a complete loss on how to reproduce this. Do you have a VM that exhibits this problem that I can take a look at?

**Comment by nmadhok** [comment]
@cachedout No sure if you'll be able to access Clemson network. Let me see if I can get you access using the account I created for @cro

**Comment by cachedout** [comment]
OK, we think we have this solved now with #28367
